account_invoice_import_invoice2data/wizard/account_invoice_import.py

```diff
@@ -76,7 +78,24 @@ def invoice2data_to_parsed_inv(self, invoice2data_res):
 
         for line in lines:
             # Manipulate line data to match with account_invoice_import
-            line['price_unit'] = float(line.get('price_unit', 0))
+            price_unit = line.get('price_unit', 0)
+
+            # Search currency information
+            currency = self.env['res.currency'].search([
+                ('name', '=', invoice2data_res.get('currency'))
+            ])
+            decimals = int(log(1 / currency.rounding, 10))
+
+            # remove thousand seperators (, OR .)
+            price_unit = re.sub(
+                r'\.(?=[\d,]*\,\d{%s}\b)' % decimals, '', price_unit)
```

Comment:
Do you need a special treatment of a currency without decimals (JPY!)

Reply:
Yes, indeed!